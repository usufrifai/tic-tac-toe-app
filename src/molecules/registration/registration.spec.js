import React from 'react';
import Registration from './index';
import {act, create} from 'react-test-renderer';
import {  Button, Form  } from 'react-bootstrap'


describe('Registration', () => {
    it('should render  Registration Component', () => {
        const component = create(
            <Registration 
            registationError={jest.fn()}
            onRegistationSubmit={jest.fn()}
            setFirstPlayer={jest.fn()}
            setSecondPlayer={jest.fn()}
            firstPlayer="Yousuf"
            secondPlayer="Rismi"
            />
          );
          const instance = component.root;
          const button = instance.findByType(Button);
          const field = instance.findAllByType(Form.Control); 
          act(() =>{
              field[0].props.onChange({ target: { value: "a" } });
              field[1].props.onChange({ target: { value: "c" } });
          });
          act(() => button.props.onClick());
          let tree = component.toJSON();
          expect(tree).toMatchSnapshot();
    });
  });