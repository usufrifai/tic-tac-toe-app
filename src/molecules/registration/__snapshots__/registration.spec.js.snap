// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Registration should render  Registration Component 1`] = `
<form
  className="registration-container"
>
  <div
    className="list-group"
    onKeyDown={[Function]}
  >
    <div
      className="list-group-item list-group-item-danger"
      data-rb-event-key={null}
      disabled={false}
      onClick={[Function]}
    >
      Please provide unique names and names cannot be empty 
    </div>
  </div>
  <div
    className="heading"
    style={
      Object {
        "width": undefined,
      }
    }
  >
    <h3>
      Register
    </h3>
  </div>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="playerOne"
    >
      <h4>
         Player One 
      </h4>
    </label>
    <input
      className="field form-control"
      id="playerOne"
      onChange={[Function]}
      placeholder="First Player Name"
      required={true}
      style={
        Object {
          "fontSize": "20px",
          "height": "50px",
        }
      }
      type="text"
      value="Yousuf"
    />
  </div>
  <div
    className="form-group"
  >
    <label
      className="form-label"
      htmlFor="playerTwo"
      style={
        Object {
          "marginTop": "5px",
        }
      }
    >
      <h4>
         Player Two 
      </h4>
    </label>
    <input
      className="form-control"
      id="playerTwo"
      onChange={[Function]}
      placeholder="Second Player Name"
      required={true}
      style={
        Object {
          "fontSize": "20px",
          "height": "50px",
        }
      }
      type="text"
      value="Rismi"
    />
  </div>
  <div
    className="form-group"
  >
    <button
      className="btn btn-warning"
      disabled={false}
      onClick={[Function]}
      style={
        Object {
          "fontSize": "20px",
          "height": "50px",
        }
      }
      type="submit"
    >
      Start Game
    </button>
  </div>
</form>
`;
